src/index.ts

```diff
@@ -1,17 +1,38 @@
-import { Dispatch, useState } from 'react';
+import { Dispatch, useCallback, useEffect, useState } from 'react';
 
 export default function useLocalStorage(
   key: string,
   initialValue: string = ''
 ): [string, Dispatch<string>] {
   const [value, setValue] = useState(
-    () => localStorage.getItem(key) || initialValue
+    () => window.localStorage.getItem(key) || initialValue
   );
 
   const setItem = (newValue: string) => {
     setValue(newValue);
     window.localStorage.setItem(key, newValue);
   };
 
+  useEffect(() => {
+    const newValue = window.localStorage.getItem(key);
+    if (value !== newValue) {
+      setValue(newValue || initialValue);
+    }
+  });
+
+  const handleStorage = useCallback(
+    (event: StorageEvent) => {
+      if (event.key === key && event.newValue !== value) {
+        setValue(event.newValue || initialValue);
+      }
+    },
+    [value]
+  );
+
+  useEffect(() => {
+    window.addEventListener('storage', handleStorage);
+    return () => window.removeEventListener('storage', handleStorage);
```

Comment:
If `value` changes, causing `handleStorage` callback to change, will `removeEventListener` still work here then? 🤔

Reply:
It will remove and append again in the next cycle.